plugins/CoreHome/angularjs/sparkline/sparkline.component.js

```diff
@@ -45,6 +45,13 @@
             };
 
             var urlParams = piwikApi.mixinDefaultGetParams($element.extend(defaultParams, vm.params));
+
+            // Append the token_auth to the URL if it was set (eg. embed dashboard)
+            var token_auth = piwik.broadcast.getValueFromUrl('token_auth');
+            if (token_auth.length) {
+                urlParams.token_auth = token_auth;
+            }
+
```

Comment:
I guess this means the token will become stored in the browser cache? I guess it can't be avoided in this case, but would it be a bit better if we only added it if in the widgetized iframe? So if someone adds the token auth to a URL when not in a widgetized whatever, it won't get added to every sparkline URL.

Reply:
Actually I copied that from here:
https://github.com/matomo-org/matomo/blob/3.x-dev/plugins/CoreHome/javascripts/sparkline.js#L31-L35

But guess makes sense to replace both places and check if `module` parameter is `Widgetize`?

Reply:
changed it

Reply:
Tested locally and noticed the token_auth is always added to the sparklines when token_auth is in the URL due to https://github.com/matomo-org/matomo/blob/3.x-dev/plugins/CoreVisualizations/Visualizations/Sparklines/Config.php#L339. Doesn't affect the single metric view widget since that is done client side. Not sure if it's worth changing this.

Reply:
The JavaScript would add it back if it's required, right?